migration/migration.go

```diff
@@ -56,47 +73,69 @@ var migrations = []Migration{
 
 var ErrCallWithNewTransaction = errors.New("call with new transaction")
 
-func MigrateIfNeeded(targetDB db.DB, network utils.Network, log utils.SimpleLogger) error {
+func MigrateIfNeededWithCallback(ctx context.Context, targetDB db.DB, network utils.Network, log utils.SimpleLogger, f func()) error {
```

Comment:
It appears this is used only for testing purposes. We should test `MigrateIfNeeded` directly if possible. 

That may require removing the `migrations` global variable. If so, one of us can open an issue about refactoring it out somehow.